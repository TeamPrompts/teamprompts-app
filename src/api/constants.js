export const ACTIVE = 'active';
export const ASC = 'asc';
export const AUTHOR = 'Author';
export const DESC = 'desc';
export const EXAMPLES = 'Examples';
export const FITB = 'FITB';
export const FITBS = 'FITBs';
export const ID = 'ID';
export const NAME = 'Name';
export const PROMPTS = 'Prompts';
export const STATUS = 'Status';
export const TAGS = 'Tags';
