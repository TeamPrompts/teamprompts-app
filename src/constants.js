// XS (4), S (8), M (16), L (32)

export const BLANK = '________________'; // M
export const BLANK_XS = '____';
export const MASTER_DETAIL_V1 = 'MASTER_DETAIL_V1';
export const NONE = 'NONE';
export const SEPARATOR = '__BLANK__';
export const TEAMPROMPTS_APP_EXPERIMENT = 'TEAMPROMPTS_APP_EXPERIMENT';

export const modes = {
  blanks: 'blanks',
  blanksXS: 'blanksXS',
  examples: 'examples',
  input: 'input',
  prompts: 'prompts'
};

export const pageTypes = {
  CategoryPage: 'CategoryPage',
  HomePage: 'HomePage',
  PromptsPage: 'PromptsPage'
};
