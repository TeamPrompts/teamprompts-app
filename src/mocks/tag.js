import tags from './tags';

const tag = tags.find((tag) => tag.id === 'recTqD26ElVvXKVad');

export default tag;
