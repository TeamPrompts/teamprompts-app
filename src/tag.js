const tag = {
  fitbs: [
    'receM4Gus7HH43oLM',
    'rec99C9lxMl6aSedq',
    'rec590aP1VRIE8f23',
    'recNsahDzKCx0IQPk'
  ],
  id: 'recTqD26ElVvXKVad',
  name: 'retrospectives'
};

export default tag; // INFO: this is a mock
