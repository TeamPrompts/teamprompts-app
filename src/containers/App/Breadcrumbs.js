import React from 'react';
import { Link } from 'react-router-dom';
import { BLANK_XS, pageTypes } from '../../constants';

function Breadcrumbs({ pageType, tag }) {
  switch (pageType) {
    case pageTypes.CategoryPage:
      return (
        <div className="mt-2">
          <Link className="text-indigo-700 underline" to="/">
            Home
          </Link>
          {' > '}
          <span className="capitalize">{tag.name}</span>
        </div>
      );
    case pageTypes.HomePage:
    default:
      return <div className="mt-2">&nbsp;</div>;
    case pageTypes.PromptsPage:
      return (
        <div className="mt-2">
          <Link className="text-indigo-700 underline" to="/">
            Home
          </Link>
          {' > '}
          <Link
            className="capitalize text-indigo-700 underline"
            to={`/${tag.slug}`}
          >
            {tag.name}
          </Link>
          {' > '}
          {BLANK_XS}
        </div>
      );
  }
}

export default Breadcrumbs;
