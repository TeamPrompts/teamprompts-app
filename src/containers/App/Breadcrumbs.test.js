import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import { create } from 'react-test-renderer';
import tag from '../../mocks/tag';
import { pageTypes } from '../../constants';
import Breadcrumbs from './Breadcrumbs';

describe('Breadcrumbs', () => {
  const scenarios = [
    {
      description: pageTypes.CategoryPage,
      props: {
        pageType: pageTypes.CategoryPage,
        tag
      }
    },
    {
      description: pageTypes.HomePage,
      props: {
        pageType: pageTypes.HomePage
      }
    },
    {
      description: pageTypes.PromptsPage,
      props: {
        pageType: pageTypes.PromptsPage,
        tag
      }
    },
    {
      description: 'no page type',
      props: {}
    }
  ];

  scenarios.forEach(({ description, props }) => {
    it(description, () => {
      const tree = create(
        <Router>
          <Breadcrumbs {...props} />
        </Router>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
