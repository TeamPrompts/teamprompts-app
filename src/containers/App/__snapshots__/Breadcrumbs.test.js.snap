// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs CategoryPage 1`] = `
<div
  className="mt-2"
>
  <a
    className="text-indigo-700 underline"
    href="#/"
    onClick={[Function]}
  >
    Home
  </a>
   &gt; 
  <span
    className="capitalize"
  >
    retrospectives
  </span>
</div>
`;

exports[`Breadcrumbs HomePage 1`] = `
<div
  className="mt-2"
>
   
</div>
`;

exports[`Breadcrumbs PromptsPage 1`] = `
<div
  className="mt-2"
>
  <a
    className="text-indigo-700 underline"
    href="#/"
    onClick={[Function]}
  >
    Home
  </a>
   &gt; 
  <a
    className="capitalize text-indigo-700 underline"
    href="#/retrospectives"
    onClick={[Function]}
  >
    retrospectives
  </a>
   &gt; 
  ____
</div>
`;

exports[`Breadcrumbs no page type 1`] = `
<div
  className="mt-2"
>
   
</div>
`;
